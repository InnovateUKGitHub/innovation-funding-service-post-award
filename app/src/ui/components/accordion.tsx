import * as React from "react";

interface Props {
  title: string;
  content: JSX.Element;
}

interface State {
  accordionOpen: boolean;
}

export class Accordion extends React.Component<Props, State> {

  constructor(props: Props) {
    super(props);
    this.state = {
      accordionOpen: false
    };
  }

  render() {
    const styles = {
      buttonStyle: {
        background: "none",
        border: "none"
      }
    };

    return (
      <div className="govuk-accordion__section">
        <div className="govuk-accordion__section-header" onClick={() => this.setState({accordionOpen: !this.state.accordionOpen})}>
          <button style={styles.buttonStyle} className="govuk-heading-m govuk-!-margin-bottom-3 govuk-!-margin-top-3">{this.props.title}</button>
          <img className="govuk-!-padding-right-2" src={this.state.accordionOpen ? "/assets/images/icon-minus.png" : "/assets/images/icon-plus.png"}/>
        </div>
        {this.state.accordionOpen && <div className="govuk-accordion__section-panel">
          {this.props.content}
        </div>}
      </div>
    );
  }
}
