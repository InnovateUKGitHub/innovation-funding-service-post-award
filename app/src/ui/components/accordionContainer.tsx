import * as React from "react";

interface Props {
  children: JSX.Element | JSX.Element[];
}

export class AccordionContainer extends React.Component<Props> {

  render() {
    return (
      <div className="govuk-accordion">
        {this.props.children}
      </div>
    );
  }
}
