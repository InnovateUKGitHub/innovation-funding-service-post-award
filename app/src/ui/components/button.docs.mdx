import { Meta, Primary, Controls, Canvas, Story } from "@storybook/blocks";
import * as ButtonStories from "./button.stories";

<Meta of={ButtonStories} />

# Button

> Related links:
>
> - [GOV.UK Design System on Button](https://design-system.service.gov.uk/components/button/)
> - [Take part in the 'Button' discusson on GitHub and share your research](https://github.com/alphagov/govuk-design-system-backlog/issues/34)

<Primary />
<Controls />

## Scenarios

### Default Buttons

<Canvas>
  <Story of={ButtonStories.PrimaryButton} />
</Canvas>

### Start Buttons

Start Buttons have not been implemented by IFSPA.

### Secondary Buttons

<Canvas>
  <Story of={ButtonStories.SecondaryButton} />
</Canvas>

### Warning Buttons

<Canvas>
  <Story of={ButtonStories.WarningButton} />
</Canvas>

### Disabled Buttons

Disabled buttons have poor contrast and can confuse some users, so avoid them if possible.

<Canvas>
  <Story of={ButtonStories.DisabledButton} />
</Canvas>

### Link Buttons

Link buttons are not GOV.UK components;
They are used in IFSPA in place of a button where buttons would be "too crowded",
such as within lists of documents, where each row requires a delete button.

Do not use this component in place of a link.
Use this component when a button would suffice.

<Canvas>
  <Story of={ButtonStories.LinkButton} />
</Canvas>
