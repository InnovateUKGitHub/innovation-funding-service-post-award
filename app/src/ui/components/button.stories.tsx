import type { Meta, StoryObj } from "@storybook/react";

import { Button } from "./styledButton";

const meta: Meta<typeof Button> = {
  title: "GOV.UK Components/Button",
  component: Button,
  argTypes: {
    styling: {
      defaultValue: "Primary",
      description: "The apperance of the button",
    },
    disabled: {
      defaultValue: false,
      description: "Whether the button is disabled",
      type: "boolean",
    },
  },
};

export default meta;

type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  tags: ["govuk/component"],
  args: {
    styling: "Primary",
    children: "Return to Desktop",
  },
};

export const PrimaryButton: Story = {
  tags: ["govuk/component"],
  args: {
    styling: "Primary",
    children: "A primary button",
  },
};

export const DisabledButton: Story = {
  tags: ["govuk/component"],
  args: {
    styling: "Primary",
    disabled: true,
    children: "A disabled button",
  },
};

export const SecondaryButton: Story = {
  tags: ["govuk/component"],
  args: {
    styling: "Secondary",
    children: "A secondary button",
  },
};

export const WarningButton: Story = {
  tags: ["govuk/component"],
  args: {
    styling: "Warning",
    children: "A warning button",
  },
};

export const LinkButton: Story = {
  tags: ["ifspa/component"],
  args: {
    styling: "Link",
    children: "A link button",
  },
};
