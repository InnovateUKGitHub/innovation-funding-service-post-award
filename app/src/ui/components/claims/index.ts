export * from "./claimDetailsLink";
export * from "./claimLastModified";
export * from "./AwardRateOverridesMessage";
export * from "./claimPeriodDate";
export * from "./claimTable";
export * from "./claimReviewTable";
export * from "./forecastTable";
