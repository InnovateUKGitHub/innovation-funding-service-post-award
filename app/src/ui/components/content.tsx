import { useContent } from "@ui/hooks";
import { ContentSelector } from "@content/content";
import { useStores } from "@ui/redux";
import { Modal } from "@ui/components/modal";
import { Section } from "@ui/components/layout";
import { SummaryList, SummaryListItem } from "@ui/components/summaryList";
import { ModalLink } from "@ui/components/links";
import { Markdown } from "./renderers/markdown";

interface IContentProps {
  value: ContentSelector;
  styles?: React.CSSProperties;
}

export const Content = ({ value, styles }: IContentProps) => {
  const stores = useStores();
  const { getResultByQuery } = useContent();

  const config = stores.config.getConfig().features;
  const { key, content, markdown } = getResultByQuery(value);

  const displayValue: string | React.ReactElement = markdown ? (
    <Markdown style={styles && { color: styles.color }} value={content} />
  ) : (
    content
  );

  const renderContentHint = () => {
    const modalId = `content-${key}`;

    return (
      <>
        &nbsp;
        <i
          aria-label="Open content modal dialogue"
          role="button"
          className="content-prompt__icon"
          tabIndex={0}
          onKeyDown={e => {
            e.preventDefault();
            e.stopPropagation();
            window.location.hash = `#${modalId}`;
          }}
        />
        <Modal id={modalId}>
          <Section title="Content hint">
            <SummaryList qa="content_info">
              <SummaryListItem label="Current value" content={displayValue} qa="current_value" />

              <SummaryListItem label="Content key" content={key} qa="current_value" />
            </SummaryList>

            <ModalLink styling="PrimaryButton" modalId={modalId} open={false}>
              Close
            </ModalLink>
          </Section>
        </Modal>
      </>
    );
  };

  return (
    <>
      {displayValue}
      {config.contentHint && renderContentHint()}
    </>
  );
};
