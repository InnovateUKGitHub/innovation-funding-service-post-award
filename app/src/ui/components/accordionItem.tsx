import * as React from "react";
import cn from "classnames";
import { AccordionContext } from "./accordion";

interface Props {
  title: string;
  qa?: string;
}

interface State {
  accordionOpen: boolean;
  focused: boolean;
}

export class AccordionItem extends React.Component<Props, State> {

  static contextType = AccordionContext;
  context!: React.ContextType<typeof AccordionContext>;

  constructor(props: Props) {
    super(props);
    this.state = {
      accordionOpen: true, // on server needs to be open
      focused: false
    };
  }

  componentDidMount() {
    this.context.subscribe();
    this.setState({ accordionOpen: false }); // once mounted on client can be initally shut
  }

  onClick() {
    const newAccordionState = !this.state.accordionOpen;
    this.context.toggle(newAccordionState);
    this.setState({ accordionOpen: newAccordionState });
  }

  onBlur() {
    this.setState({ focused: false });
  }

  onFocus() {
    this.setState({ focused: true });
  }

  componentDidUpdate(prevProps: Props, prevState: State) {
    if (this.context.allOpen && !prevState.accordionOpen) {
      this.setState({
        accordionOpen: true
      });
    }
    if (this.context.allClosed && prevState.accordionOpen) {
      this.setState({
        accordionOpen: false
      });
    }
  }

  renderTitle() {
    if (!this.context.jsEnabled) {
      return <span className="govuk-accordion__section-button">{this.props.title}</span>;
    }
    return (
      <React.Fragment>
        <button type="button" className="govuk-accordion__section-button" aria-expanded={this.state.accordionOpen} data-qa={`${this.props.qa}-button`}>{this.props.title}</button>
        <span className="govuk-accordion__icon" aria-hidden={!this.state.accordionOpen}/>
      </React.Fragment>
    );
  }

  render() {
    const topLevelClasses = cn({
      "govuk-accordion__section": true,
      "govuk-accordion__section--expanded": this.state.accordionOpen
    });
    const headerClasses = cn({
      "govuk-accordion__section-header": true,
      "govuk-accordion__section-header--focused": this.state.focused
    });
    return (
      <div className={topLevelClasses} data-qa={this.props.qa}>
        <div onBlur={() => this.onBlur()} onFocus={() => this.onFocus()} onClick={() => this.onClick()} className={headerClasses} >
          <p className="govuk-accordion__section-heading">
            {this.renderTitle()}
          </p>
        </div>
        <div className="govuk-accordion__section-content">
          {this.props.children}
        </div>
      </div>
    );
  }
}
