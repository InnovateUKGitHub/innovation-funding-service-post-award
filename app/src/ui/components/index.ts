import * as Renderers from "./renderers";
import * as Projects from "./projects";
import * as Claims from "./claims";
import * as Inputs from "./inputs";
import * as MonitoringReports from "./monitoringReports";

export * from "./accordion";
export * from "./accordionItem";
export * from "./details";
export * from "./documentList";
export * from "./documentSingle";
export * from "./errorSummary";
export * from "./form";
export * from "./layout";
export * from "./links";
export * from "./linksList";
export * from "./loading";
export * from "./logs";
export * from "./navigationArrows";
export * from "./partnersAndFinanceContacts";
export * from "./periodTitle";
export * from "./projectMember";
export * from "./styledButton";
export * from "./table";
export * from "./validationError";
export * from "./validationMessage";
export * from "./validationSummary";

export { Renderers, Projects, Claims, Inputs, MonitoringReports };
