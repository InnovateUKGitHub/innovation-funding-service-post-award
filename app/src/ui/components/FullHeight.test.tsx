import { render } from "@testing-library/react";
import { FullHeight } from "@ui/components";

describe("FullHeight Components", () => {
  const requiredProps = {
    children: <p>stub content</p>,
  };

  describe("<FullHeight.Container />", () => {
    describe("@renders", () => {
      test("as default", () => {
        const { container } = render(<FullHeight.Container {...requiredProps} />);

        expect(container.firstChild).toMatchSnapshot();
      });
    });
  });

  describe("<FullHeight.Content />", () => {
    describe("@renders", () => {
      test("as default", () => {
        const { container } = render(<FullHeight.Content {...requiredProps} />);

        expect(container.firstChild).toMatchSnapshot();
      });
    });
  });
});
