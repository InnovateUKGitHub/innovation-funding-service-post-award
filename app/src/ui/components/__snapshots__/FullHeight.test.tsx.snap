// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullHeight Components <FullHeight.Container /> @renders as default 1`] = `
<div
  class="full-height-container"
>
  <p>
    stub content
  </p>
</div>
`;

exports[`FullHeight Components <FullHeight.Content /> @renders as default 1`] = `
<div
  class="full-height-content"
>
  <p>
    stub content
  </p>
</div>
`;
